doc/api/screen_handling.rst

```diff
@@ -0,0 +1,83 @@
+.. _screen_handling_label:
+
+Screen Handling
+===============
+
+.. automodule:: simpleline.render.screen_handler
+
+The :class:`ScreenHandler` class is used to schedule
+:class:`UIScreen <simpleline.render.screen.UIScreen>` to the screen stack.
+
+Screen handling is an important part of using Simpleline library and it is recommended for user
+to get familiar with this principle. Screen handler is gateway for a developer to manage stack so
+screens can be placed to
+this stack and removed from this stack. There are multiple ways how to place screen to the stack,
+these will be described below. To remove screen from the stack the screen must be closed or it can
+be replaced by other screen. To close :class:`UIScreen <simpleline.render.screen.UIScreen>`
+a developer could call the :meth:`UIScreen.close() <simpleline.render.screen.UIScreen.close()>`
+method or a screen can be closed by a user by pressing `c` (can be disabled) to continue which will
+close the screen automatically. When a screen is closed then next top screen on the stack will be
+used.
+
+**Do not instantiate the** :class:`ScreenHandler` **class!** All methods in the
+:class:`ScreenHandler` class are class methods so the :class:`ScreenHandler` shouldn't be
+instantiated at all.
+
+The following operations can be used to schedule a screen.
+
+Schedule screen
+---------------
+
+To schedule a screen use the :meth:`ScreenHandler.schedule_screen` method.
+Scheduling a screen should be used on the first screen in your application before starting the
+event loop. Screen is added to the bottom of the screen stack and it will be visible as the last
+screen in a stack.
+
+This is the only way how to add screen to the screen stack without emitting a redraw call.
+
+.. _push_screen_label:
+
+Push screen
+-----------
+
+To push a screen to the stack use the :meth:`ScreenHandler.push_screen` method.
+Pushing screen to the screen stack will place screen to the top of the stack so this screen will be
+drawn on next draw call. However, the original screen will still be on the stack so after that new
+screen will be closed the original screen will be rendered again. If you need to avoid this behavior
+please close active screen before pushing new screen to the stack.
+
+The redraw signal will be emitted automatically when a screen is pushed.
+
+.. _push_screen_modal_label:
+
+Push screen as modal
+--------------------
+
+To push a modal screen to the stack use the :meth:`ScreenHandler.push_screen_modal` method.
+This behaves the same as :ref:`push_screen_label` with one importance difference. The pushed screen
+will act as a modal screen. Modal screen has its own even loop so event processing of the old
+loop is blocked to the time point when the modal screen will be closed.
+
+Also every other scheduling technique is not blocking call but when the
+:meth:`ScreenHandler.push_screen_modal` method is invoked then the code will be blocked until the
```

Comment:
`is invoked then` -> `is invoked, then` (and instead of code do you mean calls?)